File: selection_sort.java
//Selection sort O(n2)
//Used where memory write should be limited spave complexity O(n)
//Not much useful

class selectionsort
{
    public static int[] selectionsort(int[] arr)
    {
        for(int i=0;i<arr.length;i++)
        {
            //Find the min element
            int min = arr[i], min_index=i;
            for(int j=i+1;j<arr.length;j++)
            {
                if(arr[j] < min)
                {
                    min = arr[j];
                    min_index = j;
                }
            }
            //swap first element with min element
            int temp = arr[min_index];
            arr[min_index] = arr[i];
            arr[i] = temp;
        }
        return arr;
    }
    public static void main(String args[])
    {
        int[] arr = {3,2,1,9,6,5,0};
        arr = selectionsort(arr);
        for(int i=0;i<arr.length;i++)
        {
            System.out.println(arr[i]);
        }
    }
}

File: selection_sort.c
// C program for implementation of selection sort
#include <stdio.h>

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}
void selection_sort(int arr[], int size)
{
    for(int i=0; i<size; i++)
    {
        int min_index=i;
        for(int j=i+1; j<size; j++)
        {
            if(arr[j] < arr[min_index])
            {
                min_index = j;
            }
        }
        swap(&arr[min_index], &arr[i]);
    }
}
int main()
{
    int arr[] = {64,25,12,22,11};
    int size = sizeof(arr)/sizeof(arr[0]);
    selection_sort(arr, size);
    for(int i=0; i<size; i++)
    {
        printf("%d\n", arr[i]);
    }
}
